File: message-processor-reverse/src/main/java/message/processor/reverse/processor/MessageProcessorQueueConsumer.java
package message.processor.reverse.processor;

import message.core.log.Log;
import message.core.redis.RedisQueue;
import message.core.redis.RedisQueueConsumer;
import message.core.wrapper.MessageWrapper;

public class MessageProcessorQueueConsumer extends RedisQueueConsumer<MessageWrapper> {

    private final RedisQueue<MessageWrapper> messageDispatcherQueue;

    public MessageProcessorQueueConsumer(RedisQueue<MessageWrapper> messageProcessorQueue, RedisQueue<MessageWrapper> messageDispatcherQueue) {
        super(messageProcessorQueue);
        this.messageDispatcherQueue = messageDispatcherQueue;
    }

    @Override
    public void process(MessageWrapper messageWrapper) {
        Log.application.info("Process joke message {}", messageWrapper.getMessage().getText());
        messageWrapper.addResponse(new StringBuilder(messageWrapper.getMessage().getText()).reverse().toString());
        messageDispatcherQueue.push(messageWrapper);
    }

}

File: message-processor-echo/src/main/kotlin/message/processor/echo/processor/MessageProcessorQueueConsumer.kt
package message.processor.echo.processor

import message.core.log.Log
import message.core.redis.RedisQueue
import message.core.redis.RedisQueueConsumer
import message.core.wrapper.MessageWrapper

class MessageProcessorQueueConsumer(messageEchoProcessorQueue: RedisQueue<MessageWrapper>,
                                    private val messageDispatcherQueue: RedisQueue<MessageWrapper>) :
        RedisQueueConsumer<MessageWrapper>(messageEchoProcessorQueue) {

    override fun process(messageWrapper: MessageWrapper) {
        messageWrapper.message.text?.let {
            Log.application.info("Process echo message {}", messageWrapper.message.text)
            messageWrapper.addResponse(messageWrapper.message.text)
            messageDispatcherQueue.push(messageWrapper)
        }
    }

}
